syft/pkg/cataloger/binary/default_classifiers.go

```diff
@@ -12,7 +12,7 @@ var defaultClassifiers = []classifier{
 		EvidenceMatcher: evidenceMatchers(
 			// try to find version information from libpython shared libraries
 			sharedLibraryLookup(
-				`^libpython[0-9]+(?:\.[0-9]+)+\.so.*$`,
+				`^libpython[0-9]+(?:\.[0-9]+)+[a-z]?\.so.*$`,
```

Comment:
I see we're adding `a-z` in this capture group, but the test case was removed for `3.7.4a-vZ9` <-- Is this a case we want to capture for the lib python tests?